src/kernels/Conv_Winograd_v21_1_2_metadata.inc

```diff
@@ -146,22 +153,30 @@ __sgpr_reserve_flatscr_default = 0
     .amdhsa_dx10_clamp 0
 .end_amdhsa_kernel
 
-.ifnotdef workgroup_size_x
-    .error "workgroup_size_x must be defined"
-    .end
-.endif
-
 total_sgpr_count = .amdgcn.next_free_sgpr + 4 // vcc, xnack
 
 METADATA_WRAPPER total_sgpr_count,.amdgcn.next_free_vgpr,workgroup_size_x, <\kernel_name>
 
 .endm
 
+.macro PROLOG_KERNEL_DESCRIPTOR_WRAPPER machine_version, kernel_name_postfix
+	PROLOG_KERNEL_DESCRIPTOR miopenSp3AsmConv_v21_1_2_gfx\machine_version\()_\kernel_name_postfix
+.endm
+
+.macro EPILOG_KERNEL_DESCRIPTOR_WRAPPER machine_version, kernel_name_postfix
+	EPILOG_KERNEL_DESCRIPTOR miopenSp3AsmConv_v21_1_2_gfx\machine_version\()_\kernel_name_postfix
+.endm
+
 .macro KERNEL_PROLOG kernel_name_postfix
-	PROLOG_KERNEL_DESCRIPTOR miopenSp3AsmConv_v21_1_2_\kernel_name_postfix
+	PROLOG_KERNEL_DESCRIPTOR_WRAPPER %machine_version, \kernel_name_postfix
 .endm
 
 .macro KERNEL_EPILOG kernel_name_postfix
-	EPILOG_KERNEL_DESCRIPTOR miopenSp3AsmConv_v21_1_2_\kernel_name_postfix
+	EPILOG_KERNEL_DESCRIPTOR_WRAPPER %machine_version, \kernel_name_postfix
 .endm
 
+.set machine_version, .amdgcn.gfx_generation_number
```

Comment:
Do we really need `machine_version`? I think we can use `.amdgcn.gfx_generation_number` in `KERNEL_PROLOG` and `KERNEL_EPILOG` directly, right?

Reply:
`machine_version` also used in each shader declaration (_**Conv_Winograd_v21_1_2_xxx.s**_ files) to determine which kernel to use (gfx9 or gfx10).

Here is an [example](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/148bb845d8d557e7fa5a7dce3a9708d5bd134661/src/kernels/Conv_Winograd_v21_1_2_fp16_dot2_edc_dilation2.s#L30-L34)

Reply:
Thanks. The question is: if we always set `machine_version` to `.amdgcn.gfx_generation_number`, then why we need this symbol (`machine_version`) at all?

Reply:
This abstraction implied the ability to add COv2 support. However, at this moment the solver only supports COv3, so you are right and this symbol is not needed here.

Reply:
Correct.

NOTE: For COv2/v3 kernels, we use `.option.machine_version_major/minor/stepping` (see `rocm_version.inc`).

Reply:
Thank you for explanation. I'll keep this in mind for the future